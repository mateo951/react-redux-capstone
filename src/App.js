import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Navigation from './feature/navigation/Navigation';
import Coins from './feature/coins/Coins';
import './App.css';

function App() {
  return (
    <Router>
      <Navigation />
      <Switch>
        <Route exact path="/">
          <Coins />
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
