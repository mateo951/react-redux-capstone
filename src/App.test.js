import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from './redux/configureStore';
import App from './App';

describe('Test app snapshots', () => {
  it('testing rendering of page', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const mainHeader = screen.getByText('Cripto Live');
    fireEvent.click(mainHeader);
    expect(screen.getByText('Most Valuable -')).toMatchSnapshot();
  });
});
