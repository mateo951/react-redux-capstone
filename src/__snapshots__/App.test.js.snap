// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app snapshots testing rendering of page 1`] = `
<h3>
  Most Valuable -
   
  <p>
    Loading Crypto
  </p>
</h3>
`;
